block/manager.go

```diff
@@ -147,27 +147,37 @@ func (m *Manager) Start(ctx context.Context) error {
 		}
 	}
 
-	if !isSequencer {
-		// Fullnode loop can start before syncing from DA
-		go uevent.MustSubscribe(ctx, m.Pubsub, "applyGossipedBlocksLoop", p2p.EventQueryNewNewGossipedBlock, m.onNewGossipedBlock, m.logger)
-	}
-
 	// TODO: populate the accumulatedSize on startup
 
-	err = m.syncBlockManager()
-	if err != nil {
-		return fmt.Errorf("sync block manager: %w", err)
-	}
+	go m.refreshBlockSyncAdvertiseBlocks(ctx)
 
 	if isSequencer {
 		// Sequencer must wait till DA is synced to start submitting blobs
 		<-m.DAClient.Synced()
+		// Only sequencer must wait to be synced from DA.
+		err = m.syncBlockManager()
+		if err != nil {
+			return fmt.Errorf("sync block manager: %w", err)
+		}
 		go m.ProduceBlockLoop(ctx)
 		go m.SubmitLoop(ctx)
 	} else {
-		go m.RetrieveLoop(ctx)
-		go m.SyncToTargetHeightLoop(ctx)
+		// Full-nodes can sync from DA but it is not necessary to wait for it, since it can sync from P2P as well in parallel.
```

Comment:
tbh it's quite confusing.
we now have 5 goroutines for syncing full node
also `syncBlockManager` was to used blocking, so need to make sure no concurrency assumptions broke.

Reply:
im not sure what you propose to change here. in this pr, full-nodes will sync in parallel from da  (syncBlockManager ) and p2p, and the concurrency is protected in attemptApplyCachedBlocks (the common part of both) by retrieverMu. i think the manager part probably can be refactored and improved, but i think is better to do in other prs, since it will need to be refactored once we are done with all syncing ADRs (e.g. validating all blocks from p2p with blocks posted to da).